app/components/RemoveMachine.js

```diff
@@ -11,15 +11,17 @@ const RemoveMachine = ({ user }) => {
     return(
         <View>
             {showModal && <RemoveMachineModal closeModal={() => setShowModal(false)}/>}
-            {user.loggedIn && 
-                    <FontAwesome 
-                        name='trash' 
-                        size={30}
-                        color={'#e4606a'}
-                        style={{ marginRight: 10 }}
-                        onPress={() => setShowModal(true)}
-                    />
-            }
+            <FontAwesome
+                name='trash'
+                size={30}
+                color={'#e4606a'}
+                style={{ marginRight: 10 }}
+                onPress={() => setShowModal(true)}
+                onPress={user.loggedIn ?
+                            () => setShowModal(true) :
+                            () => this.props.navigation.navigate('Login')
```

Comment:
`this.props.navigation.navigate` is not defined in this scope
RemoveMachine is a function-based component (vs class) so there is no `this` context. Either way, navigation isn't being passed as a prop to this component. I will fix this, but the issue here is that we need to pass navigation to the component in order to use it.